job.py

```diff
@@ -0,0 +1,85 @@
+import shutil
+from urllib.request import urlopen
+from pathlib import Path
+import ssl
+import json
+from logger import logger
+
+
+def coroutine(f):
+    def wrap(*args, **kwargs):
+        gen = f(*args, **kwargs)
+        gen.send(None)
+        return gen
+    return wrap
+
+
+def get_and_write_data(condition, url):
+    context = ssl._create_unverified_context()
+    file = 'punchline.txt'
```

Comment:
имя файла потом тоже лучше перенести в параметры функции get_data:
```suggestion
    def write_data(output_file: str) -> None:
        ...
```